www/app/login/login.controller.js

```diff
@@ -39,18 +40,26 @@
         password === userInfo.password;
     }
 
-    function authenticate(username, password) {
-
-      LoginFirebaseService.getUserInfo(username)
-        .then(function(userInfo) {
-          if (_credentialsAreCorrect(username, userInfo, password)) {
-            _authSuccess();
-            RevisionService.setUsername(username);
-          }else {
-            authError();
-          }
-        });
+    function authenticate(username, password){
+      if(areEmptyCredentials(username, password)){
+        LoginFirebaseService.logIn(username)
+            .then(function(user){
+              if(user.hasOwnProperty('username') && angular.equals(user.password, password)) {
+                LoginFirebaseService.setAuthUser(username);
+                _authSuccess();
+              } else {
+                _authError();
+              }
```

Comment:
Clean code tip: create a function to handle the success of the login and then call it in the `then` of `LoginFirebaseService` like this: 
``` LoginFirebaseService.logIn(username)